rdr_service/offline/etm_duplicate_detector.py

```diff
@@ -0,0 +1,146 @@
+            .select_from(EtmQuestionnaireResponse)
+            .join(
+                duplicate_qr,
+                duplicate_qr.response_hash
+                == EtmQuestionnaireResponse.response_hash,
+            )
+            .filter(
+                EtmQuestionnaireResponse.etm_questionnaire_response_id
+                != duplicate_qr.etm_questionnaire_response_id,
+                or_(
+                    EtmQuestionnaireResponse.created < duplicate_qr.created,
+                    EtmQuestionnaireResponse.etm_questionnaire_response_id < duplicate_qr.etm_questionnaire_response_id,
+                ),
+                EtmQuestionnaireResponse.classificationType
+                == QuestionnaireResponseClassificationType.COMPLETE,
+                duplicate_qr.classificationType
+                == QuestionnaireResponseClassificationType.COMPLETE,
+            )
+        ).distinct()
+        duplicate_ids = [d_id[0] for d_id in duplicates.all()]
+        return duplicate_ids
+
+    @staticmethod
+    def mark_responses_duplicate(duplicate_ids: List[int], session) -> None:
+        """Sets the classification type for etm_questionnaire_response_ids in `duplicate_ids` as DUPLICATE"""
+        session.execute(
+            update(EtmQuestionnaireResponse)
+            .where(
+                EtmQuestionnaireResponse.etm_questionnaire_response_id.in_(duplicate_ids)
+            )
+            .values({
+                    EtmQuestionnaireResponse.classificationType: QuestionnaireResponseClassificationType.DUPLICATE
+                    })
+        )
+        session.commit()
+
+    @staticmethod
+    def clean_pdr_module_data(duplicate_responses: List[int], session, project=GAE_PROJECT) -> None:
+        """
+        Delete PDR data records (pdr_mod_* table data) that are considered orphaned if the related
+        etm_questionnaire_response_id has been flagged as a duplicate.
+        :param duplicate_responses:  List of etm_questionnaire_response_id values that have been marked as DUPLICATE
+        :param session:  A get_database().session() object
+        :param project:  project name
+        """
+
+        # For new RDR-PDR pipeline:  generate PDR delete record events for the marked duplicates.  Allow calls
+        # during unittests for mocks/param validation.  submit_pipeline_pubsub_msg() will enforce project restrictions
+        if project != "localhost" or os.environ.get("UNITTEST_FLAG", "0") == "1":
+            submit_pipeline_pubsub_msg(
+                table="etm_questionnaire_response",
+                action="delete",
+                pk_columns=["etm_questionnaire_response_id"],
+                pk_values=duplicate_responses,
+                project=project,
+            )
+
+            logging.info(
+                f"Sent PubSub notifications to mark {len(duplicate_responses)} records for deletion."
+            )
+
+        # TODO: Once the new pipeline is fully operational, the following code can be deleted:
```

Comment:
@brianmkimmel  Because the EtM data was never transferred through our old RDR-PDR pipeline and only implemented in our new pipeline, you can skip everything beyond the TODO.   That activity is still needed for regular `questionnaire_response` records that get marked as duplicate since those were part of the old pipeline, which is why it exists in the other response duplication detector code.

Reply:
@pegbertsch Thanks for explaining that. I've updated to remove that part of the code.